beacon-chain/rpc/eth/validator/handlers.go

```diff
@@ -297,3 +299,76 @@ func (s *Server) SubmitSyncCommitteeSubscription(w http.ResponseWriter, r *http.
 		cache.SyncSubnetIDs.AddSyncCommitteeSubnets(pubkey48[:], startEpoch, sub.SyncCommitteeIndices, totalDuration)
 	}
 }
+
+// ProduceSyncCommitteeContribution requests that the beacon node produce a sync committee contribution.
+func (s *Server) ProduceSyncCommitteeContribution(w http.ResponseWriter, r *http.Request) {
+	ctx := r.Context()
+	subIndex := r.URL.Query().Get("subcommittee_index")
+	index, valid := shared.ValidateUint(w, "Subcommittee Index", subIndex)
+	if !valid {
+		return
+	}
+	rawSlot := r.URL.Query().Get("slot")
+	slot, valid := shared.ValidateUint(w, "Slot", rawSlot)
+	if !valid {
+		return
+	}
+	blockRoot := r.URL.Query().Get("beacon_block_root")
+	valid = shared.ValidateHex(w, "Beacon Block Root", blockRoot)
+	if !valid {
+		return
+	}
+	contribution, ok := s.produceSyncCommitteeContribution(ctx, w, primitives.Slot(slot), index, []byte(blockRoot))
+	if !ok {
+		return
+	}
+	response := &ProduceSyncCommitteeContributionResponse{
+		Data: &shared.SyncCommitteeContribution{
+			Slot:              strconv.FormatUint(uint64(contribution.Slot), 10),
+			BeaconBlockRoot:   hexutil.Encode(contribution.BeaconBlockRoot),
+			SubcommitteeIndex: strconv.FormatUint(contribution.SubcommitteeIndex, 10),
+			AggregationBits:   hexutil.Encode(contribution.AggregationBits),
+			Signature:         hexutil.Encode(contribution.Signature),
+		},
+	}
+	http2.WriteJson(w, response)
+}
+
+// ProduceSyncCommitteeContribution requests that the beacon node produce a sync committee contribution.
+func (s *Server) produceSyncCommitteeContribution(
+	ctx context.Context,
+	w http.ResponseWriter,
+	slot primitives.Slot,
+	index uint64,
+	blockRoot []byte,
+) (*ethpbv2.SyncCommitteeContribution, bool) {
+	msgs, err := s.SyncCommitteePool.SyncCommitteeMessages(slot)
+	if err != nil {
+		http2.HandleError(w, "Could not get sync subcommittee messages: "+err.Error(), http.StatusInternalServerError)
+		return nil, false
+	}
+	if msgs == nil {
+		http2.HandleError(w, "No subcommittee messages found", http.StatusNotFound)
+		return nil, false
+	}
+	aggregatedSigAndBits, err := s.CoreService.AggregatedSigAndAggregationBits(
+		ctx,
+		&ethpbalpha.AggregatedSigAndAggregationBitsRequest{
+			Msgs:      msgs,
+			Slot:      slot,
+			SubnetId:  index,
+			BlockRoot: blockRoot,
+		},
+	)
+	if err != nil {
+		http2.HandleError(w, "Could not get contribution data: "+err.Error(), http.StatusInternalServerError)
+		return nil, false
+	}
+	return &ethpbv2.SyncCommitteeContribution{
```

Comment:
I think we should rename this function to `aggregatedSigAndBits` and return `([]byte, []byte, bool)`. I don't see the point of returning an object whose most fields are just input parameters. Also if we decide to return an object, it should be the sync committee contribution defined in `v1alpha`, not in `eth` because we want to remove all `eth` protos asap.